File: isMonotonic/isMonotonic.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Time : 2019/6/26 12:19
# @Author : hhx06
# @Site : 
# @File : isMonotonic.py
# @Software: PyCharm
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        flag = A[-1] - A[0]
        if flag > 0:
            for i in range(1, len(A)):
                if A[i] < A[i - 1]:
                    return False
        else:
            for i in range(1, len(A)):
                if A[i] > A[i - 1]:
                    return False

        return True

File: isMonotonic/isMonotonic.go
package main

import "fmt"

//如果数组是单调递增或单调递减的，那么它是单调的。
//
//如果对于所有 i <= j，A[i] <= A[j]，那么数组 A 是单调递增的。 如果对于所有 i <= j，A[i]> = A[j]，那么数组 A 是单调递减的。
//
//当给定的数组 A 是单调数组时返回 true，否则返回 false。
func isMonotonic(A []int) bool {
	len := len(A)
	if A[0] <= A[len -1]{
		for i :=1 ;i <len;i++ {
			if A[i-1] >A[i]{
				return false
			}
		}
	}else{
		for i :=1 ;i <len;i++ {
			if A[i-1] <A[i]{
				return false
			}
		}
	}
	return true
}

func main()  {
	A:= []int{1,2,2,1}
	fmt.Print(isMonotonic(A))
}



